libathemecore/users.c

```diff
@@ -21,6 +21,8 @@ static mowgli_heap_t *user_heap = NULL;
 mowgli_patricia_t *userlist;
 mowgli_patricia_t *uidlist;
 
+char hostmaskbuf[NICKLEN + USERLEN + HOSTLEN + 2];
```

Comment:
this needs an extra byte for the string terminator; i'd also add a comment explaining where the number comes from to avoid people making mistakes in the future

you could also put this as `static` inside the `format_hostmask` function i think; might be a bit cleaner?